File: 0135-candy/0135-candy.cpp
class Solution {
public:
    int candy(vector<int>& rating) {
        int n=rating.size(); 
        vector<int> ans(n,1);
        int sum=0;
        for(int i=1;i<n;i++)
        {
            if(rating[i]>rating[i-1])
            {
                ans[i]=ans[i-1]+1; 
            }
        }
        for(int i=n-1;i>0;i--)
        {
            if(rating[i]<rating[i-1])
            {
                ans[i-1]=max(ans[i-1],ans[i]+1);
            }
        }
        for(int i=0;i<n;i++)
        {
            sum+=ans[i];
        }
        return sum;
    }
};

File: 0135-candy/0135-candy.java
class Solution {
    public int candy(int[] ratings) {
        int sum=0; 
        int n=ratings.length;
        int [] arr=new int [n];
        for(int i=0;i<n;i++)
        {
            arr[i]=1;
        }
        for(int i=1;i<ratings.length;i++)
        {
            if(ratings[i-1]<ratings[i])
            {
                arr[i]=1+arr[i-1];
            }
        }
        for(int i=n-2;i>=0;i--)
        {
            if(ratings[i]>ratings[i+1])
            {
                arr[i]=Math.max(arr[i],arr[i+1]+1);
            }
        }
        for(int i=0;i<arr.length;i++)
        {
            sum+=arr[i];
        }
        return sum;
    }
}